src/main/java/net/sf/jabref/gui/preftabs/PreferencesDialog.java

```diff
@@ -239,27 +251,32 @@ public OkAction() {
 
         @Override
         public void actionPerformed(ActionEvent e) {
+            storeAllSettings();
+        }
+    }
 
-            // First check that all tabs are ready to close:
-            int count = main.getComponentCount();
-            Component[] comps = main.getComponents();
-            for (int i = 0; i < count; i++) {
-                if (!((PrefsTab) comps[i]).validateSettings()) {
-                    return; // If not, break off.
-                }
-            }
-            // Then store settings and close:
-            for (int i = 0; i < count; i++) {
-                ((PrefsTab) comps[i]).storeSettings();
-            }
-            Globals.prefs.flush();
+    class ExportAction extends AbstractAction {
 
-            setVisible(false);
-            MainTable.updateRenderers();
-            GUIGlobals.updateEntryEditorColors();
-            frame.setupAllTables();
-            frame.getGroupSelector().revalidateGroups(); // icons may have changed
-            frame.output(Localization.lang("Preferences recorded."));
+        public ExportAction() {
+            super("Export");
+        }
+
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            FileDialog dialog = new FileDialog(frame).withExtension(FileExtensions.XML);
+            dialog.setDefaultExtension(FileExtensions.XML);
+            Optional<Path> path = dialog.saveNewFile();
+
+            path.ifPresent(exportFile -> {
+                try {
+                    storeAllSettings();
+                    prefs.exportPreferences(exportFile.toString());
+                } catch (JabRefException ex) {
+                    LOGGER.warn(ex.getMessage(), ex);
+                    JOptionPane.showMessageDialog(PreferencesDialog.this, ex.getLocalizedMessage(),
+                            Localization.lang("Export preferences"), JOptionPane.ERROR_MESSAGE);
```

Comment:
The Exception should be treated consistently. Either use `LOGGER.error` or `JOptionPane.WARNING_MESSAGE`.